### ISSUE TITLE
ByteArray[ByteArrayLen, 0] not evaluated

### ISSUE THREAD

**Issue by denis-bogdanas** [kframework/kore#1649]
KEVM `kore-repl` gets stuck with the following K:
```
      <k>
        #asWord ( #buf ( 32 , ?ACCT0 ) ++ ( b"00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000002387e97500000000000000000000000000000000000000000000000000000000" [ 160 .. 0 ] ) ) ~> #push ~> #pc [ MLOAD ] ~> #execute ~> failure ""
      </k>
```
and the following new global constraint:
```
Stuck:

  #Not ( #Ceil ( b"00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000008000000000000000000000000000000000000000000000000000000000000000002387e97500000000000000000000000000000000000000000000000000000000" [ 160 .. 0 ] ) )
#And
...
```
Rules for this function:
```
    syntax ByteArray ::= ByteArray "[" Int ".." Int "]" [function]
 // --------------------------------------------------------------
    rule WS [ START .. WIDTH ] => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) requires START <Int #sizeByteArray(WS) [concrete]
    rule WS [ START .. WIDTH ] => padRightBytes(.Bytes, WIDTH, 0)                                              [owise, concrete]
```

How to reproduce: KEVM branch `symb-testing`, command `make tests/symb-test/demoSymbTest-spec.k.symb-prove`, run up to step 801, using `kore-repl`. `kore-exec` doesn't work because of other issue.

**Comment by ttuegel** [comment]
The first rule shouldn't apply in this case. I think the problem is that the backend can't prove that the `owise` case applies, either.

**Comment by ana-pantilie** [comment]
@denis-bogdanas What `K` version are you using? I've tried to run the spec with both the current `kore` `k` version and the one `evm-semantics` uses and I keep getting the following error:
```
[Error] Compiler: Attempting to split non-cell term
`<testerAcctId>`(#noDots(.KList),ACCTTO,#noDots(.KList))
  while executing phase "concretizing configuration" on sentence at
        Source(/home/ana/kore/evm-semantics/.build/defn/haskell/evm-symb-testing.k)
        Location(84,10,92,102)
        Source(/home/ana/kore/evm-semantics/.build/defn/haskell/evm-symb-testing.k)
        Location(91,10,91,47)
```

**Comment by denis-bogdanas** [comment]
@ana-pantilie I mentioned in description KEVM branch and command to run.

**Comment by ana-pantilie** [comment]
@denis-bogdanas I asked what `K` version you're using, you didn't mention that in the description.

**Comment by denis-bogdanas** [comment]
I'm actually using a command like `make <command> K_SUBMODULE=/mnt/d/k5` to pass a different version of K. Using latest K+latest Kore.

**Comment by ttuegel** [comment]
I have a suspicion that this is actually related to #1637. I suggest we try this again when #1659 is merged.

**Comment by denis-bogdanas** [comment]
This also causes slow Haskell execution due to the constraint mentioned above. I fixed it by modifying ByteArray[_ .._] rules to:
```
    rule WS [ START .. WIDTH ] => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) 
        requires START <Int #sizeByteArray(WS) andBool WIDTH =/=Int 0   [concrete]
    rule WS [ START .. WIDTH ] => padRightBytes(.Bytes, WIDTH, 0)                                              
        requires START >=Int #sizeByteArray(WS)  andBool WIDTH =/=Int 0 [concrete]
    rule WS [ START .. 0 ] => .ByteArray [concrete]
```

**Comment by ttuegel** [comment]
We found that the `owise` rule does not apply because the frontend (incorrectly) includes the left-hand side of a `simplification` lemma in its `requires` clause, and we are approaching the issue in the frontend.

**Comment by ttuegel** [comment]
Fixed by kframework/k#1182.
